### PR TITLE
chore: Unify ID extraction regex from browser url

### PR DESCRIPTION
**Pull request by abhvsn**
## Description
Regex update to make it compatible to extract identifiers for both Mongo ObjectIds and UUIDs. This will help us to keep the unified logic required in `pg` branch.

## Automation

/ok-to-test tags="@tag.Datasource, @tag.GenerateCRUD, @tag.ImportExport, @tag.Fork, @tag.Workspace, @tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9405554200>
> Commit: 3959703aab1d10e28d3b80057793476467126929
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9405554200&attempt=2" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Enhanced URL path handling to support both UUID and Mongo ObjectIds.

- **Refactor**
  - Replaced hardcoded page IDs with dynamic variables across multiple test files for improved maintainability and flexibility.

- **Tests**
  - Updated test cases to use dynamic page IDs, ensuring consistency and easier updates in the future.

### CODE REVIEW

File: app/client/src/ce/constants/routes/appRoutes.ts [hunk 1]
```diff
@@ -7,12 +7,19 @@ import { matchPath } from "react-router";
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const { match } = require("path-to-regexp");
 
+// Regex to extract the id from the URL path which supports both the formats:
+// 1. With UUID
+// 2. With Mongo ObjectIds
+const MONGO_OBJECT_ID_REGEX = "[0-9a-fA-F]{24}";
+const UUID_REGEX =
+  "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}";
```
Comment (sharat87) [inline app/client/src/ce/constants/routes/appRoutes.ts:15]: Are we expecting upper-case here? UUIDs are case-insensitive by nature so I'd rather we accept only lower-case letters here, unless we specifically expect upper-case letters to show up here. What do you think of only accepting `[0-9a-f]` for the hex characters?

Comment (sharat87) [reply app/client/src/ce/constants/routes/appRoutes.ts:15]: Also, the comments 1 and 2 are in the reverse order of the regexes. 🙂

**Review by sharat87 — COMMENTED** [review]
Client tests seem to be failing, is that okay?

**Comment by abhvsn** [comment]
Checking test failures in EE with PR: https://github.com/appsmithorg/appsmith-ee/pull/4373

**Comment by abhvsn** [comment]
@sharat87 on EE as well we have all greens, can you take a look when you get time.

**Review by sharat87 — APPROVED** [review]

**Pull request merged**